### PR TITLE
Datasource: Fix missing raw SQL query in Query Inspector when query returns zero rows

### PR DESCRIPTION
**Pull request by baldm0mma**
**What is this feature?**

When a query successfully returns an empty result set (frame.Rows() == 0), the previous behavior was to (unintentionally) hide the raw SQL queryString in the QueryInspector.tsx, due to no fields being present. This fix - on the return of an empty result - simply ads back a single empty field slice so that the FE can once again display the raw SQL query.

**Why do we need this feature?**

So QueryInspector.tsx works the way a user would expect.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes [this](https://github.com/grafana/support-escalations/issues/5799) escalation

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [na] If this is a pre-GA feature, it is behind a feature toggle.
- [na] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/mssql/mssql_test.go [hunk 1]
```diff
@@ -1263,6 +1263,50 @@ func TestMSSQL(t *testing.T) {
...
+		}
+		err = sess.CreateTable(emptyObj{})
+		require.NoError(t, err)
+
+		t.Run("When no rows are returned, should return an empty frame", func(t *testing.T) {
+			query := &backend.QueryDataRequest{
+				Queries: []backend.DataQuery{
+					{
+						JSON: []byte(`{
+							"rawSql": "SELECT empty_key, empty_val FROM empty",
```
Comment (mmandrus) [inline pkg/tsdb/mssql/mssql_test.go:1287]: Heads up - this should be `FROM empty_obj` to match the struct name change. Same with `postgres_test.go` if you wanted to make that `emptyObj` as well.

File: pkg/tsdb/sqleng/sql_engine.go [hunk 2]
```diff
@@ -314,7 +314,8 @@ func (e *DataSourceHandler) executeQuery(query backend.DataQuery, wg *sync.WaitG
 
 	// If no rows were returned, no point checking anything else.
 	if frame.Rows() == 0 {
-		queryResult.dataResponse.Frames = data.Frames{}
+		frame.Fields = []*data.Field{}
```
Comment (zoltanbedi) [inline pkg/tsdb/sqleng/sql_engine.go:317]: I'd love to see a comment on this why this is needed but other than that good job!

Comment (baldm0mma) [reply pkg/tsdb/sqleng/sql_engine.go:317]: Great point; I agree; a change like this is hard to understand without a little context.

Comment (baldm0mma) [reply pkg/tsdb/sqleng/sql_engine.go:320]: Updated with annotation snippet.

**Review by mmandrus — APPROVED** [review]
LGTM 💪

**Review by zoltanbedi — APPROVED** [review]
Good job! Love the comment in the code. 😍

**Pull request merged**